enzyme/Enzyme/MLIR/Dialect/EnzymeOps.td

```diff
@@ -145,6 +145,26 @@ def AutoDiffOp : Enzyme_Op<"autodiff",
   let hasCanonicalizer = 1;
 }
 
+def AutoDiffRegionOp : Enzyme_Op<"autodiff_region", [AffineScope]> {
```

Comment:
I'd also make it a alloca memory op scope (I forget the name @ftynse if you recall). I might also remove the affine scope.